File: socialgo/src/main/java/com/fungo/socialgo/social/listener/OnLoginListener.java
package com.fungo.socialgo.social.listener;

import com.fungo.socialgo.social.exception.SocialError;
import com.fungo.socialgo.social.model.LoginResult;

/**
 * CreateAt : 2016/12/25
 * Describe : 登陆监听
 *
 * @author chendong
 */
public interface OnLoginListener {

    void onStart();

    void onSuccess(LoginResult loginResult);

    void onCancel();

    void onFailure(SocialError e);
}

File: socialgo_core/src/main/java/com/fungo/socialgo/listener/OnLoginListener.kt
package com.fungo.socialgo.listener

import com.fungo.socialgo.exception.SocialError
import com.fungo.socialgo.model.LoginResult

/**
 * 登录回调
 */
interface OnLoginListener {

    fun getFunction(): FunctionListener

    fun onStart(start: () -> Unit) {
        getFunction().onStart = start
    }

    fun onSuccess(success: (result: LoginResult) -> Unit) {
        getFunction().onLoginSuccess = success
    }

    fun onCancel(cancel: () -> Unit) {
        getFunction().onCancel = cancel
    }

    fun onFailure(failure: (error: SocialError) -> Unit) {
        getFunction().onFailure = failure
    }

}
